client/planning-extension/src/assignments-overview/index.tsx

```diff
@@ -0,0 +1,245 @@
+}
+
+function fetchAssignments(userId: IUser['_id']): Promise<IState['assignments']> {
+    return queryRawJson<IRestApiResponse<IAssignmentItem>>(
+        'assignments',
+        {
+            page: '1',
+            sort: '[("planning.scheduled", 1)]',
+            source: JSON.stringify({
+                query: {
+                    bool: {
+                        must: [
+                            {term: {'assigned_to.user': userId}},
+                            {terms: {'assigned_to.state': ['assigned', 'submitted', 'in_progress']}},
+                        ],
+                    },
+                },
+            }),
+        },
+    ).then(({_items}) => {
+        superdesk.dispatchEvent(
+            'menuItemBadgeValueChange',
+            {menuId: 'MENU_ITEM_PLANNING_ASSIGNMENTS', badgeValue: _items.length.toString()},
+        );
+
+        return _items;
+    });
+}
+
+export class AssignmentsList extends React.PureComponent<{}, {loading: true} | IState> {
+    private eventListenersToRemoveBeforeUnmounting: Array<() => void>;
+
+    constructor(props: {}) {
+        super(props);
+
+        this.state = {loading: true};
+
+        this.eventListenersToRemoveBeforeUnmounting = [];
+
+        this.handleContentChanges = this.handleContentChanges.bind(this);
+
+        this.eventListenersToRemoveBeforeUnmounting.push(
+            addWebsocketMessageListener(
+                'resource:created',
+                (event) => {
+                    const {resource, _id} = event.detail.extra;
+
+                    this.handleContentChanges([{changeType: 'created', resource: resource, itemId: _id}]);
+                },
+            ),
+        );
+
+        this.eventListenersToRemoveBeforeUnmounting.push(
+            addWebsocketMessageListener(
+                'resource:updated',
+                (event) => {
+                    const {resource, _id, fields} = event.detail.extra;
+
+                    this.handleContentChanges([{
+                        changeType: 'updated',
+                        resource: resource,
+                        itemId: _id,
+                        fields: fields,
+                    }]);
+                },
+            ),
+        );
+
+        this.eventListenersToRemoveBeforeUnmounting.push(
+            addWebsocketMessageListener(
+                'resource:deleted',
+                (event) => {
+                    const {resource, _id} = event.detail.extra;
+
+                    this.handleContentChanges([{changeType: 'deleted', resource: resource, itemId: _id}]);
+                },
+            ),
+        );
+    }
+
+    handleContentChanges(changes: Array<IResourceChange>) {
+        const state = this.state;
+
+        if (state.loading === true) {
+            return;
+        }
+
+        const {assignments} = state;
+
+        const refetchDesks = changes.find(({resource}) => resource === 'desks');
```

Comment:
I guess this only makes sense if new desk is created, but on update there is no need to fetch them all

Reply:
Would it be beneficial to have an interface in the `superdeskApi` to  **fetch and subscribe** to resources?

Where `resource` is required, and `itemId` is optional, and the implementation can reside in Superdesk (internally subscribing to websocket notifications)?

This would help later on when we add a caching store (or something the like of). The `superdeskApi` interface would not change, but all uses of it would automatically get cached results. Otherwise we would have to come back here and implemen caching here as well

Reply:
Good point. I'm not sure how a subscribe method would work, it seems it would need a react wrapper the same way redux needs one. I have instead added a few higher order components that fetch data and keep it up to date. When we have cache we can connect it to the implementation of these HOCs.